op_crates/url/lib.rs

```diff
@@ -102,41 +101,42 @@ pub fn op_url_parse(
         .unwrap_or_default()
     ))
   })?;
-  Ok(json!(UrlParts {
-    href: quirks::href(&url),
-    hash: quirks::hash(&url),
-    host: quirks::host(&url),
-    hostname: quirks::hostname(&url),
-    origin: &quirks::origin(&url),
-    password: quirks::password(&url),
-    pathname: quirks::pathname(&url),
-    port: quirks::port(&url),
-    protocol: quirks::protocol(&url),
-    search: quirks::search(&url),
-    username,
-  }))
+  Ok(UrlParts {
+    href: quirks::href(&url).to_string(),
```

Comment:
These allocation were there before, they were just implicit, see my notes:
> This also applies to places where `json!()` offered some "useful" implicit conversions for references (e.g: &str, Cow, ...) to owned types (that can be returned and serialized), to my understanding there was no (performance) upside to returning refs, since json!() would convert them to owned types anyway

You'll see that `serde_json::Value` only has owned `String`s:
https://sourcegraph.com/github.com/serde-rs/json@d735585370d3d62502130051529410464e147224/-/blob/src/value/mod.rs#L111-171

Reply:
Ah... very painful. Good to know.

Reply:
Just to elaborate on specifics:
- string refs get serialized [here in the json macro](https://sourcegraph.com/github.com/serde-rs/json@d735585370d3d62502130051529410464e147224/-/blob/src/macros.rs#L275-280)
- Which calls to `to_value()`
- Which ultimately calls this [serialize_str()](https://sourcegraph.com/github.com/serde-rs/json@d735585370d3d62502130051529410464e147224/-/blob/src/macros.rs#L275-280) method of the serializer
- Which is basically `Ok(Value::String(value.to_owned()))`